src/query/plan/operator.cpp

```diff
@@ -5148,6 +5148,8 @@ class CallProcedureCursor : public Cursor {
   bool stream_exhausted{true};
   bool call_initializer{false};
   std::optional<std::function<void()>> cleanup_{std::nullopt};
+  std::shared_ptr<procedure::Module> module_{};
+  const mgp_proc *proc_{nullptr};
```

Comment:
This would be a breaking change in my opinion. If we cache, this means the user can't reload the module. It also means that such queries that reload a module will block the the memgraph's other queries that want to take a shared lock and execute queries with procedures. I am against this change honestly